CONTRIBUTING_MODELS.md

```diff
@@ -0,0 +1,235 @@
+
+- [ ] An `xx/xx_presets.py` file with links to weights uploaded to a personal GCP bucket/Google Drive [[Example Link](https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/models/distil_bert/distil_bert_presets.py)].
+- [ ] An `xx/xx_presets_test.py` file with runnable tests for each preset [[Example Link](https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/models/distil_bert/distil_bert_presets_test.py)].
+- [ ] A `tools/checkpoint_conversion/convert_xx_checkpoints.py` which is reusable script for converting checkpoints [[Example Link](https://github.com/keras-team/keras-nlp/blob/master/tools/checkpoint_conversion/convert_distilbert_checkpoints.py)].
+- [ ] A Colab notebook showing an end-to-end task such as text classification, etc. The task model can be built using the backbone model, with the task head on top [[Example Link](https://gist.github.com/mattdangerw/bf0ca07fb66b6738150c8b56ee5bab4e)].
+
+### Step 5: PR #4 and Beyond - Add XX Tasks and Preprocessors
+
+This PR is optional and is for adding tasks like classifier, masked LM, etc. [[Example Link](https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/models/distil_bert/distil_bert_classifier.py)]
+
+## Detailed Instructions
+
+This section discusses, in details, every necessary step.
+
+### Step 1: Open an issue/Find an open issue
+
+Before getting started with the code, it's important to check if there are any
+[open issues](https://github.com/keras-team/keras-nlp/issues?q=is%3Aissue+is%3Aopen+label%3Amodel-contribution)
+related to the model you wish to contribute. If there is an open issue, you can
+claim it by commenting on the issue and letting us know that you're interested
+in working on it. This helps us keep track of who is working on what and avoid
+duplicated effort.
+
+If there aren't any open issues, you can create one by clicking the "New Issue"
+button on our repository page.
+
+Note that you need not have all the answers or complete knowledge of the inner
+workings of the model at the time of opening the issue. But it is appreciated if
+you can furnish as much detail as possible to enable us to help you with the
+contribution! 🙂
+
+### Step 2: PR #1 - Add XXBackbone
+
+#### Add the backbone class
+
+Once you are done identifying all the required layers, you should write down the
+model backbone class.
+
+To keep the code simple and readable, we follow
+[Keras' functional style model](https://keras.io/guides/functional_api/) wrapped
+around by a class to implement our models.
+
+A model is typically split into three/four sections. We would recommend you to
+compare this side-by-side with the
+[`keras_nlp.layers.DistilBertBackbone` source code](https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/models/distil_bert/distil_bert_backbone.py)!
+
+**Inputs to the model**
+
+Generally, the standard inputs to any text model are:
+    - `token_ids`: tokenised inputs (An integer representation of the text sequence).
```

Comment:
The formatting looks messed up here (at least on the github UI), probably don't need the large indent for the list?